packages/bvaughn-architecture-demo/src/suspense/GenericCache.ts

```diff
@@ -0,0 +1,54 @@
+import { Record, STATUS_PENDING, STATUS_REJECTED, STATUS_RESOLVED, Wakeable } from "./types";
+import { createWakeable } from "../utils/suspense";
+
+export function createGenericCache<TParams extends Array<any>, TValue>(
```

Comment:
This is clever!

Kind of a nit, but can we rename this file to `createGenericCache.ts` since it's a cache-factory and not actually a cache itself?